src/domain/tradingview/__tests__/utils.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+import { getCurrentCandleTime } from "../utils";
+
+describe("utils", () => {
+  // Asia/Dubai 4 pm 19th Nov 2024
+  const currentDate = new Date("2024-11-19T16:00:00+04:00");
+
+  const currentTimeSeconds = Math.floor(currentDate.getTime() / 1000);
+
+  beforeEach(() => {
+    vi.useFakeTimers();
+    vi.setSystemTime(currentDate);
+  });
+
+  afterEach(() => {
+    vi.useRealTimers();
+  });
+
+  describe("getCurrentCandleTime", () => {
+    it("should return the current period start", () => {
+      const result = getCurrentCandleTime("1m");
+      expect(result).toBe(currentTimeSeconds);
+
+      // elapse 1s
+      vi.advanceTimersByTime(1000);
```

Comment:
maybe also check 1000+1? it should be next candle, right?

Reply:
Because the period here is 1m the candles are withing a 60s groups so i test 1 second to be first candle and then 60s more (61s from beginning) so it gives the next candle. Giving 1001 ms would still give first candle i think.